### PR TITLE
MINOR: improve state directory test

### PR DESCRIPTION
**Pull request by mjsax**
We get a failing Jenkins build with
```
java.lang.AssertionError
at org.junit.Assert.fail(Assert.java:86)
at org.junit.Assert.assertTrue(Assert.java:41)
at org.junit.Assert.assertFalse(Assert.java:64)
at org.junit.Assert.assertFalse(Assert.java:74)
at org.apache.kafka.streams.processor.internals.StateDirectoryTest.shouldNotCreateBaseDirectory(StateDirectoryTest.java:363)
```

This is an attempt to stabilize the test plus some cleanup.

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StateDirectoryTest.java [hunk 1]
```diff
@@ -48,11 +49,14 @@
...
     private File stateDir;
-    private String applicationId = "applicationId";
+    private final String applicationId = "applicationId";
     private StateDirectory directory;
     private File appDir;
 
-    private void initializeStateDirectory(final boolean createStateDirectory) {
+    private void initializeStateDirectory(final boolean createStateDirectory) throws Exception {
+        if (stateDir != null) {
+            cleanup();
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StateDirectoryTest.java:59]: This is the actual fix.

**Comment by mjsax** [comment]
Call for review @guozhangwang @bbejeck @vvcephei

**Review by bbejeck — APPROVED** [review]

**Review by guozhangwang — APPROVED** [review]

**Comment by mjsax** [comment]
Retest this please

**Review by kamalcph — COMMENTED** [review]
@mjsax 
LGTM, thanks for fixing the issue!

**Comment by mjsax** [comment]
I just updated this PR, to delete the directory _after_ a new one is picked. Seems more reasonable. Will merge after green build if no objections raised.

**Comment by guozhangwang** [comment]
> I just updated this PR, to delete the directory _after_ a new one is picked. Seems more reasonable. Will merge after green build if no objections raised.

+1

**Comment by mjsax** [comment]
Test failed last night again. Let's see if this fixes the issue...

**Pull request merged**